realm/realm-library/src/androidTest/kotlin/io/realm/rule/BlockingLooperThread.kt

```diff
@@ -88,7 +89,14 @@ class BlockingLooperThread {
      * Runs the test on a Looper thread
      */
     fun runBlocking(threadName: String = "TestLooperThread", emulateMainThread: Boolean = false, test: () -> Unit) {
-        RunInLooperThreadStatement(threadName, emulateMainThread, test).evaluate()
+        RunInLooperThreadStatement(threadName, emulateMainThread, test).evaluate(false)
+    }
+
+    /**
+     * Runs the test on a Looper thread. Returns an object that can be used to wait for the test to complete
+     */
+    fun runDetached(threadName: String = "TestLooperThread", emulateMainThread: Boolean = false, test: () -> Unit): Condition? {
```

Comment:
Nice 💯

Reply:
But why is the return value `Condition?`. It always returns the condition and it would clean up some `?` used in the test code.